src/app/window-handler.ts

```diff
@@ -918,6 +918,28 @@ export class WindowHandler {
         delete this.windows[key];
     }
 
+    /**
+     * Executes the given command via a child process
+     *
+     * @param util {string}
+     * @param utilArgs {ReadonlyArray<string>}
+     */
+    public execCmd(util: string, utilArgs: ReadonlyArray<string>): Promise<ChildProcess> {
+        logger.info(`window handler: execCmd: util: ${util} utilArgs: ${utilArgs}`);
+        return new Promise<ChildProcess>((resolve, reject) => {
+            return execFile(util, utilArgs, (error: ExecException | null) => {
+                if (error) {
+                    logger.info(`window handler: execCmd: error: ${error}`);
+                }
+                if (error && error.killed) {
+                    // processs was killed, just resolve with no data.
+                    return reject(error);
+                }
```

Comment:
we should add a comment explaining why we have separated out the `if` statements.